src/index.ts

```diff
@@ -325,20 +327,21 @@ class Storyblok {
 		resolveId: string
 	): void {
 		if (fields.indexOf(`${jtree.component}.${treeItem}`) > -1) {
-			if (typeof jtree[treeItem] === 'string') {
-				if (this.relations[resolveId][jtree[treeItem]]) {
-					jtree[treeItem] = this._cleanCopy(
-						this.relations[resolveId][jtree[treeItem]]
+			const cleanCopyStory = (uuid: string) => {
```

Comment:
everything works like a charm now! good job @ademarCardoso !
I would suggest to change this function name in something more significant. For what I understood this could return the uuid of the story if not included in the stories to resolve otherwise returns the story object, so maybe something like `getStoryReference` can clarify it more, what do you think?

Reply:
also, I was thinking here 🤔 Maybe we can extrapolate this function outside a private one like this:

```
	/**
	 *
	 * @param resolveId A counter number as a string
	 * @param uuid The uuid of the story
	 * @returns string | object
	 */
	private getStoryReference(resolveId: string, uuid: string): string | object {
		console.log(this.relations, resolveId, uuid)
		if (!this.relations[resolveId][uuid]) return uuid
		if (!this.stringifiedStoriesCache[uuid])
			this.stringifiedStoriesCache[uuid] = JSON.stringify(
				this.relations[resolveId][uuid]
			)
		return JSON.parse(this.stringifiedStoriesCache[uuid])
	}
```

and then call it this way later on:

line 340: `this.getStoryReference(resolveId, jtree[treeItem])`
line 343: `this.getStoryReference(resolveId, uuid)`

Reply:
You are right Lori, i will try to do that